test/src/main/java/org/apache/accumulo/test/functional/ExitCodesIT_SimpleSuite.java

```diff
@@ -0,0 +1,329 @@
+          serverClass = TabletServer.class;
+          methodName = "updateIdleStatus";
+          ctorParams = new Class<?>[] {ConfigOpts.class, Function.class, String[].class};
+          ctorArgs = new Object[] {new ConfigOpts(), new ServerContextFunction(),
+              tabletServerArgs.toArray(new String[] {})};
+          break;
+        case MONITOR:
+        case ZOOKEEPER:
+        default:
+          throw new UnsupportedOperationException(st + " is not currently supported");
+      }
+
+      final Implementation implementation;
+      switch (behavior) {
+        case ERROR:
+          implementation =
+              ExceptionMethod.throwing(StackOverflowError.class, "throwing unknown error");
+          break;
+        case EXCEPTION:
+          implementation =
+              ExceptionMethod.throwing(RuntimeException.class, "throwing runtime exception");
+          break;
+        case SHUTDOWN:
+          implementation =
+              MethodCall.invoke(AbstractServer.class.getMethod("requestShutdownForTests"));
+          break;
+        default:
+          throw new UnsupportedOperationException(behavior + " is not currently supported");
+      }
+
+      // Dynamically create the subclass with the specified behavior, load it
+      // into the JVM, and return it's Class.
+      // creates a subclass of server class defining a public constructor that
+      // takes an Integer, but calls the server class constructor with the
+      // appropriate args. Calls to the intercepted method name are intercepted
+      // by the supplied implementation
+      final Class<? extends AbstractServer> dynamicServerType =
+          new ByteBuddy().subclass(serverClass, ConstructorStrategy.Default.NO_CONSTRUCTORS)
+              .name("org.apache.accumulo.test.functional.exit_codes." + serverClass.getSimpleName()
+                  + "_" + behavior.name())
+              .defineConstructor(Visibility.PUBLIC).withParameters(Integer.class)
+              .intercept(MethodCall.invoke(serverClass.getDeclaredConstructor(ctorParams)).onSuper()
+                  .with(ctorArgs))
+              .method(ElementMatchers.named(methodName)).intercept(implementation).make()
+              .load(serverClass.getClassLoader()).getLoaded();
+
+      // Find and invoke the constructor of the dynamically created class
+      final Constructor<? extends AbstractServer> ctor =
+          dynamicServerType.getDeclaredConstructor(Integer.class);
+      final AbstractServer dynamicServerInstance = ctor.newInstance(1);
+
+      // Start the server process
+      AbstractServer.startServer(dynamicServerInstance, LoggerFactory.getLogger(serverClass));
+    }
+
+  }
+
+  @BeforeAll
+  public static void beforeTests() throws Exception {
+    // Start MiniCluster so that getCluster() does not
+    // return null,
+    SharedMiniClusterBase.startMiniCluster();
+    getCluster().getClusterControl().stop(ServerType.GARBAGE_COLLECTOR);
```

Comment:
I don't think you can do this as part of the SimpleSuite tests. This method would only get run when running it as a standalone test. I don't think it would be called, and thus the GC wouldn't be stopped, when run as part of the suite.